common/src/main/java/com/wynntils/models/items/annotators/game/GearAnnotator.java

```diff
@@ -34,6 +48,109 @@ public ItemAnnotation getAnnotation(ItemStack itemStack, StyledText name) {
 
         GearInstance gearInstance =
                 matcher.group("unidentified") != null ? null : Models.Gear.parseInstance(gearInfo, itemStack);
+
+        if (gearInfo.tier() == GearTier.SET) {
+            Optional<SetInfo> setInfo = Optional.of(Models.Set.getSetInfoForItem(gearInfo.name()));
+            int wynnCount = getWynncraftCount(setInfo.get().name());
+            int equippedItemSlot = Models.PlayerInventory.getEquippedItemSlot(itemStack);
+            if (equippedItemSlot >= 0) {
+                Optional<SetInstance> setInstance = Optional.of(new SetInstance(
+                        setInfo.get(), getActiveItems(setInfo.get().name()), wynnCount));
+                GearSlot slot;
+                if (gearInfo.type() == GearType.RING) {
+                    slot = equippedItemSlot % 9 == 0 ? GearSlot.RING1 : GearSlot.RING2;
+                } else {
+                    slot = GearSlot.fromGearType(gearInfo.type());
+                }
+                Models.Set.updateSetInstance(slot, setInstance.get());
+            }
+        }
```

Comment:
And with the changes done in the previous suggestions, you can remove this whole part. GearInfo already has the set info from "upstream", and gear instance should be populated "upstream" too.

Reply:
Aaand, that means `getActiveItems` and `getWynncraftCount` goes as well.